core/sail/inferencer/src/main/java/org/eclipse/rdf4j/sail/inferencer/fc/FastRdfsForwardChainingSailConnection.java

```diff
@@ -0,0 +1,1026 @@
+            public Object[] toArray() {
+                return new Object[0];
+            }
+
+            @Override
+            public <T> T[] toArray(T[] a) {
+                return null;
+            }
+
+            @Override
+            public boolean add(Statement statement) {
+                return false;
+            }
+
+            @Override
+            public boolean remove(Object o) {
+                return false;
+            }
+
+            @Override
+            public boolean containsAll(Collection<?> c) {
+                return false;
+            }
+
+            @Override
+            public boolean addAll(Collection<? extends Statement> c) {
+                return false;
+            }
+
+            @Override
+            public boolean removeAll(Collection<?> c) {
+                return false;
+            }
+
+            @Override
+            public boolean retainAll(Collection<?> c) {
+                return false;
+            }
+
+            @Override
+            public void clear() {
+
+            }
+        };
+    }
+
+
+    protected int applyRules(Model model) throws SailException {
+
+
+        return 0;
+    }
+
+
+    public void addStatement(Resource subject, IRI predicate, Value object, Resource... contexts) throws SailException {
+        addStatement(true, subject, predicate, object, contexts);
+    }
+
+
+    public void addStatement(boolean actuallyAdd, Resource subject, IRI predicate, Value object, Resource... resources) throws SailException {
+
+        final boolean[] inferRdfTypeSubject = {false};
+        final boolean[] inferRdfTypeObject = {false};
+
+        if (fastRdfsForwardChainingSail.schema == null) {
+            statementCollector(fastRdfsForwardChainingSail.getValueFactory().createStatement(subject, predicate, object));
+        }
+
+        if (fastRdfsForwardChainingSail.useAllRdfsRules) {
+            addInferredStatement(subject, RDF.TYPE, RDFS.RESOURCE);
+
+            if (object instanceof Resource) {
+                addInferredStatement((Resource) object, RDF.TYPE, RDFS.RESOURCE);
+            }
+        }
+
+
+        if (predicate.getNamespace().equals(RDF.NAMESPACE) && predicate.getLocalName().charAt(0) == '_') {
+
+            try {
+                int i = Integer.parseInt(predicate.getLocalName().substring(1));
+                if (i >= 1) {
+                    addInferredStatement(subject, RDFS.MEMBER, object);
+
+                    addInferredStatement(predicate, RDF.TYPE, RDFS.RESOURCE);
+                    addInferredStatement(predicate, RDF.TYPE, RDFS.CONTAINERMEMBERSHIPPROPERTY);
+                    addInferredStatement(predicate, RDF.TYPE, RDF.PROPERTY);
+                    addInferredStatement(predicate, RDFS.SUBPROPERTYOF, predicate);
+                    addInferredStatement(predicate, RDFS.SUBPROPERTYOF, RDFS.MEMBER);
+
+                }
+            } catch (NumberFormatException e) {
+                // Ignore exception.
```

Comment:
This is actually a feature. It matches the special rdf lists that use :a rdf:_1 :item1.

So if there is an exception, it just means that someone has a property with rdf as namespace and _ as the first character but the rest of the property isn't a number. In which case we don't need to do anything.